PublicComponentCode/ComponentCode/src/java/main/ccsu/edu/grovepicomponents/LightSensor.java

```diff
@@ -0,0 +1,131 @@
+package ccsu.edu.grovepicomponents;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import edu.ccsu.error.IncompatibleSensorError;
+import edu.ccsu.interfaces.Iterator;
+import edu.ccsu.utility.CommonConstants;
+
+/**
+ * Class can access light sensor and 
+ * get the light intensity
+ */
+public class LightSensor extends Sensor {
+
+	private List<LightSensorData> sensorData;
+	/**
+	 * Creates a concrete LightSensor object
+	 * @param name
+	 * @param portNumber
+	 */
+	public LightSensor(String name, String portNumber) {
+		this.name = name;
+		this.portNumber = portNumber;
+		this.sensorData = new ArrayList<>();
+		this.sensorFile = CommonConstants.LIGHTSENSOR;
+	}
+	
+	/**
+	 * Method that takes data string, parses it, creates LightSensorData objects,
+	 * and adds them to list
+	 * @param data
+	 */
+	@Override
+	protected String addToList(String data) {
+		StringBuilder builder = new StringBuilder();
+		String[] dataToAdd = data.split(",");
+		for(String str: dataToAdd) {
+			String[] makeIntoData = str.split(" ");
+			String sensorValue = makeIntoData[0];
+			String voltage = makeIntoData[1];
+			String watts = makeIntoData[2];
+			Date date = new Date();
+			builder.append("SensorValue: " + sensorValue + " Voltage: " + voltage + " Watts: " + watts + " Date: " + date + "\n");
+			//value from output will be three numbers, first is sensorValue second is voltage, third is watts
+			sensorData.add(new LightSensorData(Integer.parseInt(sensorValue), Float.parseFloat(voltage), Float.parseFloat(watts), date));
+		}
+		return builder.toString();
+	}
+    
+	@Override
+	protected boolean checkPort(String portNumber) {
+		if(!this.getPortNumber().contains("A")) {
+			System.out.println("Must use a digital port starting with A");
+			return false;
+		}
+		return true;
+	}
+	
+	@Override
+	public void setNextSensor(Sensor nextSensor) throws IncompatibleSensorError  {
+		if(nextSensor instanceof LightSensor) {
```

Comment:
In both this and the temp and humidity sensor you reject it if the passed sensor is an instance other than its own type.  The limitation of this is a developer wouldn't realize the mistake until runtime when the IncompatibleSensorError is thrown.  A better way to handle this would be with Generics where this method would be dynamically bound to the type the developer was using so they could see at compile time that what they were trying to do was illegal.